grading/load_config_json.cpp

```diff
@@ -20,6 +20,40 @@ void json_set_default(nlohmann::json &whole_config, std::string field, const T&
   }
 }
 
+uint64_t parse_file_size(std::string size_str) {
+  // Trim whitespaces
+  size_str.erase(std::remove_if(size_str.begin(), size_str.end(), isspace), size_str.end());
+
+  // Convert string to lowercase, note that it cannot handle UTF-8 characters
+  for (char& c : size_str)
+    c = std::tolower(c);
+
+  // Extract numeric file size, will cause exception if the number is larger than 64-bit
+  size_t unit_begin = 0;
+  uint64_t file_size = std::stoull(size_str, &unit_begin);
+  // check if it is 0 or contains invalid chars     ====whitelist====
+  if (file_size == 0 || size_str.find_first_not_of("01234567890bkmgti") != std::string::npos) {
+    std::cout << "Got parsed value " << file_size << " from string: " << size_str << std::endl;
+    throw std::invalid_argument("Wrong file size");
+  }
+
+  const std::string suffices = "bkmgt";
+  // Calculate l-shifts
+  size_t lshift = 0;
+  if (unit_begin != size_str.size() // handle strings without unit, treat as byte
+      && (lshift = suffices.find(size_str[unit_begin])) == std::string::npos) {
```

Comment:
I think something like `1234 bkmgt` can be parsed through here. It is should give an error.

Reply:
This will be parsed as `1234`, as only `b` is considered as the unit.

Reply:
Yes but shouldn’t this throw an error because the input has an invalid suffix